clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4081,7 +4084,9 @@ void handleFailure(Throwable throwable) {
     }
 
     private void handleNotControllerError(AbstractResponse response) throws ApiException {
-        if (response.errorCounts().containsKey(Errors.NOT_CONTROLLER)) {
+        // When sending requests directly to the follower controller, it might return NOT_LEADER_OR_FOLLOWER error.
+        if (response.errorCounts().containsKey(Errors.NOT_CONTROLLER) ||
+                metadataManager.usingBootstrapControllers() && response.errorCounts().containsKey(Errors.NOT_LEADER_OR_FOLLOWER)) {
             handleNotControllerError(Errors.NOT_CONTROLLER);
```

Comment:
Should we pass `NOT_LEADER_OR_FOLLOWER` instead of `NOT_CONTROLLER` when it encounters the error `NOT_LEADER_OR_FOLLOWER`?

Reply:
Agree. Updated.